### ISSUE TITLE
Architectural optimizations on the gesture-toolbox

### ISSUE THREAD

**Issue by myjeeed** [Microsoft/pxt#2576]
We need to delay load the user interface dependencies for the gesture-toolbox which are: `npm msr` (MIT License), and `npm d3` (BSD 3 License).

**Comment by abchatra** [comment]
@myjeeed Is this particular to gesture-toolbox branch?

**Comment by microsoftsam** [comment]
Nope, this is an optimization that can be done for any component that we don't need for all targets / all the time.
